build/parseSpec.c

```diff
@@ -180,13 +180,15 @@ static int expandMacrosInSpecBuf(rpmSpec spec, int strip)
     if (lbuf[0] == '#')
 	isComment = 1;
 
-    lbuf = xstrdup(spec->lbuf);
+    lbuf = spec->lbuf;
 
-    rc = expandMacros(spec, spec->macros, spec->lbuf, spec->lbufSize);
-    if (rc) {
+    spec->lbuf = expandMacros(spec, spec->macros, lbuf);
```

Comment:
Here you rewrite spec->lbuf but the memory at which spec->lbuf pointed previously is not deallocated anywhere.